### ISSUE TITLE
Acc Tests: Issue with Lock/Unlock test

### ISSUE THREAD

**Issue by jtopjian** [gophercloud/gophercloud#1222]
It appears the [`TestServersActionLock`](https://github.com/gophercloud/gophercloud/blob/master/acceptance/openstack/compute/v2/servers_test.go#L440) test is incorrect.

When attempting to delete the locked server, the following error is _expected_:

```
convenience.go:35: Failure in servers_test.go, line 455: unexpected error "Expected HTTP response code [202 204] when accessing [DELETE https://example.com:8774/v2/9aa5f9f66b4b417d84d778a23acdf45b/servers/23a3dbf8-2f04-4f1a-b515-bd96f4e80543], 
but got 409 instead\n{\"conflictingRequest\": {\"message\": \"Instance 23a3dbf8-2f04-4f1a-b515-bd96f4e80543 is locked\", \"code\": 409}}"
```

However, the test is asserting that the error is nil.

**Comment by jtopjian** [comment]
It looks like an admin user is able to delete a server even if it's locked. Since the OpenLab tests are run as admin, the server is prematurely deleted. It succeeds from time to time because the second Delete call happens before the server has been completely deleted.

**Comment by jtopjian** [comment]
Fixed in #1223
